BraveWallet/Crypto/Stores/AssetDetailStore.swift

```diff
@@ -53,26 +56,28 @@ class AssetDetailStore: ObservableObject {
     assetRatioService: BraveWalletAssetRatioService,
     keyringService: BraveWalletKeyringService,
     rpcService: BraveWalletJsonRpcService,
+    walletService: BraveWalletBraveWalletService,
     txService: BraveWalletTxService,
     blockchainRegistry: BraveWalletBlockchainRegistry,
-    token: BraveWallet.BlockchainToken
+    token: BraveWallet.BlockchainToken,
+    currencyFormatter: NumberFormatter
   ) {
     self.assetRatioService = assetRatioService
     self.keyringService = keyringService
     self.rpcService = rpcService
+    self.walletService = walletService
     self.txService = txService
     self.blockchainRegistry = blockchainRegistry
     self.token = token
+    self.currencyFormatter = currencyFormatter
 
     self.keyringService.add(self)
     self.rpcService.add(self)
     self.txService.add(self)
-  }
-
-  static let priceFormatter = NumberFormatter().then {
-    $0.numberStyle = .currency
-    $0.currencyCode = "USD"
-    $0.maximumFractionDigits = 4
+    
+    walletService.defaultBaseCurrency { [self] currencyCode in
+      self.currencyFormatter.currencyCode = currencyCode
```

Comment:
dont you need a `update()` here?

Reply:
You know what, I don't think I even need these updates for fetching the default base currency in the initializers anymore now that we are observing changes in `CryptoStore` and re-using the same `currencyFormatter: NumberFormatter`. The observer will catch the change and update accordingly.